### PR TITLE
Stop attempting to cache all subresources in Ansible Operator proxy

### LINKED ISSUE #3052: In Ansible operator module k8s_log can not get Pod log
## Bug Report

**What did you do?**

In Ansible operator, use k8s_log module of Ansible Kubernetes collection to get specific Pod log failed.

Same Ansible role works well out of the operator.
Same Ansible role works well inside the operator when manually run the roles with ansible-runner cmd.(After "oc exec -ti ....." to the Pod)

**What did you expect to see?**
Return correct Pod log.

**What did you see instead? Under which circumstances?**
The return of k8s_log module was just the definition/spec of the Pod,but not the Pod log.

**Environment**
* operator-sdk version:

```
$ operator-sdk  version
operator-sdk version: "v0.15.2", commit: "ffaf278993c8fcb00c6f527c9f20091eb8dd3352", go version: "go1.13.3 linux/amd64"
```

* go version:

```
$ go version
go version go1.14 linux/amd64
```

* Kubernetes version information:

v1.14.6+0a21dd3b3

* Kubernetes cluster kind: 
Openshift 4.2

* Are you writing your operator in ansible, helm, or go?

Ansible

**Possible Solution**

**Additional context**

### PR DESCRIPTION
**Pull request by fabianvf**
**Description of the change:**
The Ansible Operator proxy now only will attempt to pull the status subresource from the cache, all other subresources will skip the cache.

**Motivation for the change:**
The cache responds with the content of the resource, not the actual subresource request (which generally can't be watched and therefore can't be cached). This prevents actions like retrieving logs from succeeding with the expected results (instead of logs, you'd just get the API resource definition)

Fixes #3052

### CODE REVIEW

File: pkg/ansible/proxy/cache_response.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func (c *cacheResponseHandler) ServeHTTP(w http.ResponseWriter, req *http.Reques
 			break
 		}
```
Comment (camilamacedo86) [inline pkg/ansible/proxy/cache_response.go:68]: missing the fragment with bug fix

File: test/ansible/playbooks/subresources.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@
                 args:
                   - "/bin/sh"
                   - "-c"
-                  - "while true;do date;sleep 5;done"
+                  - "while true ; do echo '{{ log_message }}' ; sleep 5 ; done"
```
Comment (tengqm) [inline test/ansible/playbooks/subresources.yml:24]: Just our of curiosity ... the `log_message` here is from the `logMessage` of the CR?

Comment (fabianvf) [reply test/ansible/playbooks/subresources.yml:24]: that's correct

File: hack/tests/e2e-ansible-molecule.sh [hunk 3]
```diff
@@ -14,6 +14,7 @@ pip3 install --user pyasn1==0.4.7 pyasn1-modules==0.2.6 idna==2.8 ipaddress==1.0
 pip3 install --user molecule==3.0.2
 pip3 install --user ansible-lint yamllint
 pip3 install --user docker openshift jmespath
+ansible-galaxy collection install community.kubernetes
```
Comment (camilamacedo86) [inline hack/tests/e2e-ansible-molecule.sh:17]: That was what was missing :-) 👍

**Pull request merged**